bindings/nodejs/establish-flags.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# This file is part of the Soletta Project
+#
+# Copyright (C) 2015 Intel Corporation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in
+#     the documentation and/or other materials provided with the
+#     distribution.
+#   * Neither the name of Intel Corporation nor the names of its
+#     contributors may be used to endorse or promote products derived
+#     from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# By default, build soletta using defaults
+SOLETTA_CFLAGS="-I$(pwd)/build/soletta_sysroot/usr/include/soletta"
+SOLETTA_LIBS="-L$(pwd)/build/soletta_sysroot/usr/lib -lsoletta -Wl,-rpath $(pwd)/build/soletta_sysroot/usr/lib"
+BUILD_SOLETTA="true"
+
+# If we're running from soletta's build process, establish the prefix, use
+# local include files, and link with no rpath
+if test "x${SOLETTA_FROM_MAKE}x" == "xtruex"; then
+	PREFIX="$(cat .config | grep '^PREFIX=' | sed -r 's/^[^"]*"([^"]*)"/\1/')"
+	PREFIX="${PREFIX##/}"
+
+	SOLETTA_CFLAGS="-I$(pwd)/build/soletta_sysroot/${PREFIX}/include/soletta"
+	SOLETTA_LIBS="-L $(pwd)/build/soletta_sysroot/${PREFIX}/lib -lsoletta"
+	BUILD_SOLETTA="false"
+
+# Otherwise, try to find soletta via pkg-config
+elif pkg-config --exists soletta > /dev/null 2>&1; then
```

Comment:
I cannot use ```$SOLETTA_BUILD_DIR/lib/pkgconfig/soletta.pc``` because that file contains information that is valid only after soletta is installed. Since ```$SOLETTA_BUILD_DIR/lib/pkgconfig/soletta.pc``` refers to the prefix where soletta *will be* installed, those places are still empty during build time, so, when I try to use the includedir or the libdir given by ```$SOLETTA_BUILD_DIR/lib/pkgconfig/soletta.pc``` during the npm build process, it will fail.